File: Problems/1624. Largest Substring Between Two Equal Characters/1624. Largest Substring Between Two Equal Characters.java
import java.util.HashMap;

class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        int res = -1;
        HashMap<Character, Integer> record = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (!record.containsKey(ch)) record.put(ch, i);
            else res = Math.max(res, i-record.get(ch)-1);

        }

        return res;
    }
}

File: Problems/1624. Largest Substring Between Two Equal Characters/1624. Largest Substring Between Two Equal Characters.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        record = {}
        res = -1
        
        for i in range(len(s)):
            if s[i] in record: res = max(res, i-record[s[i]]-1)
            else: record[s[i]] = i
        
        return res